state-chain/pallets/cf-elections/src/electoral_systems/composite.rs

```diff
@@ -401,182 +304,156 @@ macro_rules! generate_electoral_system_tuple_impls {
     };
     (@ $($previous:ident,)*;: $($electoral_system:ident,)*) => {};
     (@ $($previous:ident,)*; $current:ident, $($remaining:ident,)*: $($electoral_system:ident,)*) => {
-        impl<$($electoral_system: ElectoralSystem<ValidatorId = ValidatorId>,)* ValidatorId: MaybeSerializeDeserialize + Parameter + Member, H: Hooks<$($electoral_system),*> + 'static, BorrowEA: Borrow<EA>, EA: ElectionReadAccess<ElectoralSystem = Composite<($($electoral_system,)*), ValidatorId, H>>> ElectionReadAccess for CompositeElectionAccess<tags::$current, BorrowEA, EA> {
+
+        impl<'a, $($electoral_system: ElectoralSystem<ValidatorId = ValidatorId>,)* ValidatorId: MaybeSerializeDeserialize + Parameter + Member, H: Hooks<$($electoral_system),*> + 'static, StorageAccess: RunnerStorageAccessTrait<ElectoralSystemRunner = CompositeRunner<($($electoral_system,)*), ValidatorId, StorageAccess, H>> + 'static> ElectionReadAccess for DerivedElectionAccess<tags::$current, $current, StorageAccess> {
             type ElectoralSystem = $current;
 
             fn settings(&self) -> Result<$current::ElectoralSettings, CorruptStorageError> {
-                let ($($previous,)* settings, $($remaining,)*) =self.ea.borrow().settings()?;
+                let ($($previous,)* settings, $($remaining,)*) = StorageAccess::electoral_settings_for_election(*self.id.unique_monotonic())?;
                 Ok(settings)
             }
             fn properties(&self) -> Result<$current::ElectionProperties, CorruptStorageError> {
-                match self.ea.borrow().properties()? {
+                match StorageAccess::election_properties(self.id.with_extra(CompositeElectionIdentifierExtra::$current(*self.id.extra())))? {
                     CompositeElectionProperties::$current(properties) => {
                         Ok(properties)
                     },
                     _ => Err(CorruptStorageError::new())
                 }
             }
             fn state(&self) -> Result<$current::ElectionState, CorruptStorageError> {
-                match self.ea.borrow().state()? {
+                match StorageAccess::election_state(*self.id.unique_monotonic())? {
```

Comment:
The naive way of doing this would be to read everything in and then just provide back the individual fields, which is obviously a lot less efficient when we don't need everything. Doing it lazily requires a little more complexity in creating the "Election" that's returned, in order to make this simpler, so I don't think it's an obvious win.